src/xdp-app-info.c

```diff
@@ -108,18 +110,29 @@ xdp_app_info_initialize (XdpAppInfo *app_info,
                          const char *app_id,
                          const char *instance,
                          int         pidfd,
-                         GAppInfo   *gappinfo,
+                         const char *desktop_file,
                          gboolean    supports_opath,
                          gboolean    has_network,
                          gboolean    requires_pid_mapping)
 {
   XdpAppInfoPrivate *priv = xdp_app_info_get_instance_private (app_info);
+  g_autoptr(GAppInfo) gappinfo = NULL;
+
+  if (desktop_file != NULL)
+    {
+      g_autoptr(GDesktopAppInfo) desktop_appinfo = NULL;
+
+      desktop_appinfo = g_desktop_app_info_new (desktop_file);
+      if (desktop_appinfo)
```

Comment:
do we actually need this dance of assigning the value first to `desktop_appinfo`?

Reply:
The casts don't like getting NULL and there is no guarantee that we'll be able to find the desktop file that we "detected" so I don't see how.

Reply:
oh, i didn't know that.